core/extension/extension_api_dump.cpp

```diff
@@ -884,11 +885,18 @@ Dictionary GDExtensionAPIDump::generate_extension_api() {
 						d2["hash"] = method->get_hash();
 
 						Vector<uint32_t> compat_hashes = ClassDB::get_method_compatibility_hashes(class_name, method_name);
+						Array compatibility;
 						if (compat_hashes.size()) {
-							Array compatibility;
 							for (int i = 0; i < compat_hashes.size(); i++) {
 								compatibility.push_back(compat_hashes[i]);
 							}
+						}
+
+#ifndef DISABLE_DEPRECATED
+						GDExtensionCompatHashes::get_legacy_hashes(class_name, method_name, compatibility);
+#endif
+
+						if (compatibility.size() > 0) {
 							d2["hash_compatibility"] = compatibility;
 						}
```

Comment:
General question: with `DISABLE_DEPRECATED`, do we still write this key?
Might very well be "yes", just wanted to double-check 🙂

Reply:
Hm, well, if all compatibility methods are also wrapped with `#ifdef DISABLE_DEPRECATED` then we won't have anything else to put into "hash_compatibility". But I think this is fine? The _idea_ of compatibility hashes still exists with `DISABLE_DEPRECATED`, we just aren't registering any.